File: windows/Scanner/Rules/AESEXT001.cs
using AIExposureScanner.Scanner.Models;

namespace AIExposureScanner.Scanner.Rules;

public sealed class AESEXT001 : IRule
{
    public string Id => "AES-EXT-001";
    public Severity Severity => Severity.High;

    public IReadOnlyList<Finding> Evaluate(ScanFacts facts) =>
        facts.Extensions
            .Where(extension => extension.HasTerminalAccess)
            .Select(extension => this.Finding(extension.AppId, extensionId: extension.ExtensionId))
            .ToArray();
}

File: macos/Scanner/Sources/Scanner/Rules/AESEXT001.swift
public struct AESEXT001: Rule {
    public let id = "AES-EXT-001"
    public let severity: Severity = .high

    public init() {}

    public func evaluate(_ facts: ScanFacts) -> [Finding] {
        facts.extensions.compactMap { ext in
            guard ext.hasTerminalAccess else {
                return nil
            }
            return finding(app: ext.appId, extensionId: ext.extensionId)
        }
    }
}
